src/plugins/discover/public/components/discover_grid_flyout/use_flyout_actions.tsx

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React from 'react';
+import {
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiHideFor,
+  EuiIconTip,
+  EuiText,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { useDiscoverCustomization } from '../../customizations';
+import { UseNavigationProps, useNavigationProps } from '../../hooks/use_navigation_props';
+
+interface FlyoutActionProps {
+  onClick: React.MouseEventHandler<Element>;
+  href: string;
+}
+
+const staticViewDocumentItem = {
+  id: 'viewDocument',
+  enabled: true,
+  Content: () => <ViewDocument />,
+};
+
+export const useFlyoutActions = (navigationProps: UseNavigationProps) => {
+  const { dataView } = navigationProps;
+  const { singleDocHref, contextViewHref, onOpenSingleDoc, onOpenContextView } =
+    useNavigationProps(navigationProps);
+
+  const flyoutCustomization = useDiscoverCustomization('flyout');
+
+  const {
+    viewSingleDocument = { disabled: false },
+    viewSurroundingDocument = { disabled: false },
+  } = flyoutCustomization?.actions?.defaultActions ?? {};
+  const customActions = [...(flyoutCustomization?.actions?.getActionItems?.() ?? [])];
```

Comment:
Could you add a unit test for this one too?

Reply:
Added a test right now, thanks for reviewing again this work 👍